nparse.py

```diff
@@ -89,9 +89,9 @@ def __init__(self, *args):
 
     def _load_parsers(self):
         self._parsers_dict = {
-            "maps": parsers.Maps(),
-            "spells": parsers.Spells(),
-            "discord": parsers.Discord(),
+            "maps": parsers.Maps("maps"),
```

Comment:
Yeah this is what I was talking about in discord -- the Maps class should know the name is "maps", it shouldn't need to be told its own name -- but depends on what this is USED for I guess? if it's not displayed and is just a unique key, then that's fine -- what would happen if we replaced that string with a UUID, would it change anything?

Reply:
ParserWindow now maps that to self.name.

Previously self.name was set to an empty string "" and ParserWindow never had any knowledge of self.name because it was being set by its consumer, so no init could be used for that var name. By setting it at instantiation we can then setup init methods that work on it. Like moving the set title logic into the ParserWindow instead of the consumer app Spells/Maps/Discord having to duplicate any logic that depended on self.name

Reply:
I'd rather remove name entirely from the init() and just have ParserWindow have an abstract property name() that we implement in each subclass.

Reply:
I'm ok with that, I prefer the instantiation method myself, but we can set a property if you prefer. Both methods achieve the same goal currently and if we run into an issue later we can simply refactor then.

I'd like to throw a ValueError exception in the ParserWindow if they never implemented the property then and give the valueerror a custom message to let them clearly know they need to implement a property to use the class.

I'll work on changing it up a little later on :)

Reply:
Thank you for the pushback last night and the suggestions! really helped today to re tackle the issue. I learned that we could have simply set the self.name in the consumers prior to calling super init and it did the behavior we were after, so i basically put self.name back to how it was before my changes, and simply reordered where it was happening :)